### PR TITLE
Refactor async helper

### PR DESCRIPTION
**Pull request by eugenk**
This removes the `AsyncHelper`. By circumventing some Sidekiq mechanics, this module over-complicates the code and reduces performance.

After this refactoring, every asynchronous call is done according to Sidekiq's best practices.

Additionally, we should finally get rid of the "Ontology indefinitely pending" problem because we don't schedule a parsing task to be executed at a defined point in time but rather use Sidekiq's `perfom_async` as we are supposed to do.

--

###### Todo if this gets merged:
* [ ] Add issue for the `FIXME` comment: When parsing an Ontology, we wait a second for the SQL query to finish. This was also done before this refactoring, but it is not an appropriate solution.

### CODE REVIEW

**Comment by eugenk** [comment]
I rebased this on staging

File: lib/dsl/repo.rb [hunk 1]
```diff
@@ -61,6 +61,6 @@ def save
     r.user = User.first
     r.url_maps = @url_maps
     r.save!
-    r.async :convert_to_local!
+    RepositoryCeonversionWorker.perform_async(r.id)
```
Comment (SGraef) [inline lib/dsl/repo.rb]: Here is a typo

Comment (eugenk) [inline lib/dsl/repo.rb]: Fixed in a7ce8a0

File: spec/models/repository/git_spec.rb [hunk 2]
```diff
@@ -14,12 +14,12 @@
...
 
     it 'should create a bulk job on a queue' do
-      expect { repository.remote_send('clone') }.
+      expect { repository.fetch('clone') }.
         to change(OntologyBatchParseWorker.jobs, :size)
     end
 
     it 'should run the ontology import on the cloned repository', process_jobs_synchronously: true do
-      expect { repository.remote_send('clone') }.
+      expect { repository.fetch('clone') }.
```
Comment (eugenk) [inline spec/models/repository/git_spec.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: spec/models/repository/git_spec.rb [hunk 3]
```diff
@@ -14,12 +14,12 @@
     end
 
     it 'should create a bulk job on a queue' do
-      expect { repository.remote_send('clone') }.
+      expect { repository.fetch('clone') }.
```
Comment (eugenk) [inline spec/models/repository/git_spec.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/repository_pulling_worker.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class RepositoryPullingWorker
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+
+  sidekiq_options retry: false, queue: 'default'
```
Comment (eugenk) [inline lib/repository_pulling_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/repository_conversion_worker.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class RepositoryConversionWorker < BaseWorker
+  sidekiq_options retry: false, queue: 'default'
```
Comment (eugenk) [inline lib/repository_conversion_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/repository_fetching_worker.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class RepositoryFetchingWorker < BaseWorker
+  sidekiq_options retry: false, queue: 'default'
```
Comment (eugenk) [inline lib/repository_fetching_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/oops_request_worker.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class OopsRequestWorker < BaseWorker
+  sidekiq_options retry: 3, queue: 'default'
```
Comment (eugenk) [inline lib/oops_request_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/ontology_parsing_worker.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  sidekiq_options retry: 3, queue: 'hets'
+
+  # parse_mode can be 'parse_fast' or 'parse_full'
+  # *args only holds the array files_to_parse_afterwards
+  def perform(ontology_version_id, parse_mode, *args)
+    # We need to wait a second for the SQL query to finish.
+    # Otherwise there is no OntologyVersion with id=ontology_version_id.
+    # FIXME This needs to be handled properly.
+    sleep 1 unless defined?(Sidekiq::Testing) && Sidekiq::Testing.inline?
+    super('record', 'OntologyVersion', parse_mode, ontology_version_id, *args,
```
Comment (eugenk) [inline lib/ontology_parsing_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: lib/ontology_parsing_worker.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+class OntologyParsingWorker < Worker
+  sidekiq_options retry: 3, queue: 'hets'
```
Comment (eugenk) [inline lib/ontology_parsing_worker.rb]: No thanks. You should stop suggesting this as soon as #1570 gets merged.

File: spec/lib/repository_pulling_worker_spec.rb [hunk 10]
```diff
@@ -1,20 +1,22 @@
...
-describe SidetiqWorker do
+describe RepositoryPullingWorker do
   subject! { create :repository_with_empty_remote }
-  before{ Worker.clear }
+  before { RepositoryFetchingWorker.clear }
 
   shared_examples 'perform' do |state, minutes, created_jobs_count|
-    describe "state #{state}, imported " << (minutes ? "#{minutes} minutes ago" : "never before") do
+    context("state #{state}, imported " +
+            (minutes ? "#{minutes} minutes ago" : 'never before')) do
```
Comment (eugenk) [inline spec/lib/repository_pulling_worker_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/lib/repository_pulling_worker_spec.rb [hunk 11]
```diff
@@ -1,20 +1,22 @@
...
       before do
         subject.update_attributes!(
-          {state: state.to_s, imported_at: (minutes ? minutes.minutes.ago : nil)},
-          {without_protection: true}
+          {state: state.to_s,
+           imported_at: (minutes ? minutes.minutes.ago : nil)},
```
Comment (eugenk) [inline spec/lib/repository_pulling_worker_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/lib/repository_pulling_worker_spec.rb [hunk 12]
```diff
@@ -1,20 +1,22 @@
...
 
   shared_examples 'perform' do |state, minutes, created_jobs_count|
-    describe "state #{state}, imported " << (minutes ? "#{minutes} minutes ago" : "never before") do
+    context("state #{state}, imported " +
+            (minutes ? "#{minutes} minutes ago" : 'never before')) do
       before do
         subject.update_attributes!(
-          {state: state.to_s, imported_at: (minutes ? minutes.minutes.ago : nil)},
-          {without_protection: true}
+          {state: state.to_s,
```
Comment (eugenk) [inline spec/lib/repository_pulling_worker_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/lib/repository_pulling_worker_spec.rb [hunk 13]
```diff
@@ -1,20 +1,22 @@
...
+           imported_at: (minutes ? minutes.minutes.ago : nil)},
+           without_protection: true,
```
Comment (eugenk) [inline spec/lib/repository_pulling_worker_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

**Comment by SGraef** [comment]
:+1:

**Pull request merged**